src/main/java/practice/CandidateValidator.java

```
@@ -1,5 +1,27 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.Objects;
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    @Override
+    public boolean test(Candidate candidate) {
```

Silently returning `false `on malformed data may hide bad inputs. You could throw an `IllegalArgumentException `on invalid format so that callers know the data was wrong.